fil_fungible_token/src/token/mod.rs

```diff
@@ -237,7 +237,12 @@ where
         operator: &Address,
         delta: &TokenAmount,
     ) -> Result<TokenAmount> {
-        let delta = validate_amount(delta, "allowance delta", self.granularity)?;
+        if delta.is_negative() {
+            return Err(TokenError::InvalidNegative {
+                name: "allowance delta",
+                amount: delta.clone(),
+            });
+        }
```

Comment:
Just to confirm, did you remove the granularity check here because the FRC says so? Ok.

I think we should pull this out to a `validate_allowance` method so it's clear that this isn't an oversight. I can see some well-meaning future person trying to change this back.